File: programmers 문제 모음/연습문제/탐욕법(Greedy)/체육복.py
def solution(n, lost, reserve):
    answer = n - len(lost) # 현재 체육복을 안잃어버린 학생 수 answer에 저장

     # lost, reserve에 모두 있는 학생은 제외하기
    temp = []
    for i in lost:
        if i in reserve:
            reserve.pop(reserve.index(i))
            temp.append(i)
            answer += 1
    while temp:
        lost.pop(lost.index(temp.pop()))

    # 정렬 후 체육복 빌려줄 수 있는 경우 구하기
    lost.sort()
    reserve.sort()
    reserve_check = [True] * len(reserve) # 이미 빌려준 체육복인지 확인하는 리스트
    for l in range(len(lost)):
        for r in range(len(reserve)):
            if reserve_check[r] and (lost[l]-1 == reserve[r] or lost[l] +1 == reserve[r]):
                reserve_check[r] = False
                answer += 1
                break
    return answer


# n = 5
# lost = [4, 2, 3]
# reserve = [3]
# print(solution(n, lost, reserve))

File: programmers/연습문제/탐욕법(Greedy)/체육복.js
function solution(n, lost, reserve) {
    let answer = n - lost.length;

    // 여벌의 체육복을 가져온 학생이 도난당한 경우 찾기
    for (let i = 0; i < reserve.length; i ++) {
        if (lost.includes(reserve[i])) {
            lost[lost.indexOf(reserve[i])] = -1

            reserve[i] = -1
            answer += 1
        }
    }
    // 여벌 체육복을 도난당한 학생들에게 빌려주는 경우 찾기 
    for (let r = 0; r < reserve.length; r ++) {
        if (reserve[r] === -1) { 
            continue
        }
        for (let l = 0; l < lost.length; l ++) {
            if (lost[l] === -1) {
                continue
            }
            else if (lost[l] === reserve[r] - 1 || lost[l] === reserve[r] + 1) {
                    answer += 1
                    lost[l] = -1
                    break

            }
        }
    }
    return answer;
}


// let n = 7
// let lost = [2, 3, 4]
// let reserve = [1, 2, 3, 6]
// console.log(solution(n, lost, reserve))